admin-portal/src/pages/needUpload/needUpload.tsx

```diff
@@ -1,59 +1,55 @@
 import React, { useState } from "react";
-import { useNavigate } from "react-router";
 import { PageSelection } from "../../types/pages";
 import { Page } from "layout/page";
 import "./needUpload.css";
 import { AidPackageService } from "apis/services/AidPackageService";
-import { ToastContainer, toast } from "react-toastify";
-import "react-toastify/dist/ReactToastify.css";
+import { toast } from "react-toastify";
 
 export function NeedUpload() {
   const [file, setFile] = useState(null);
+  const [fileName, setFileName] = useState("");
   const [errorList, setErrorList] = useState([]);
-  const navigate = useNavigate();
 
   function handleChange(event: any) {
     setErrorList([]);
     setFile(event.target.files[0]);
+    setFileName(event.target.value);
   }
 
   const handleSubmit = async (event: any) => {
     event.preventDefault();
-
     const formData = new FormData();
 
     if (file != undefined) {
       formData.append("file", file);
-      AidPackageService.postNeeds(formData)
-        .then((response) => {
-          toast.success(response.data, {
-            position: "top-right",
-            autoClose: 5000,
-            hideProgressBar: true,
-            closeOnClick: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined,
-          });
-
-          navigate("/");
-        })
-        .catch((error) => {
-          toast.error("File upload unsuccessful", {
-            position: "top-right",
-            autoClose: 5000,
-            hideProgressBar: true,
-            closeOnClick: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined,
-          });
-
-          const data = error.response.data
-            .split(/\r?\n/)
-            .filter((element: any) => element);
-          setErrorList(data);
+      try {
+        const response = await AidPackageService.postNeeds(formData);
+        toast.success(response.data, {
+          position: "top-right",
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
         });
+        setFileName("");
+      } catch (e: any) {
```

Comment:
Don't use `any`.

Reply:
what can i use here instead?

Reply:
It says
> Catch clause variable type annotation must be 'any' or 'unknown' if specified.

Reply:
It should be: 

```
 } catch (error) {
      if (error instanceof AxiosError) {
```

Reply:
If the response comes with a response, then you should create an interface and provide it using generics:

ex: 
error instanceof AxiosError<X>

Reply:
it should be 

```
catch (e){
if (!Object.prototype.hasOwnProperty.call(error, 'isAxiosError')) {
     const error = error as AxiosError<Y>;
}
}
```

`Y` should be the type of response payload.

Reply:
Apparently the `Object.prototype` doesn't have the attribute `isAxiosError`. `AxiosError` has that property. So this actually returns false. Found out theres an easier way to do this. 

> catch (e) {
        if (axios.isAxiosError(e)) {
          const error = e as AxiosError<string>;
        }
   }